R/plotColTile.R

```diff
@@ -0,0 +1,206 @@
+    retrieve_search <- switch(type,
+                              "row" = "rowData",
+                              "column" = "colData")
+    if(!is.null(factor_x_by)){
+        faxtor_x_by_out <- retrieve_FUN(object, factor_x_by,
+                                        search = retrieve_search)
+    }
+    if(!is.null(factor_y_by)){
+        faxtor_y_by_out <- retrieve_FUN(object, factor_y_by,
+                                        search = retrieve_search)
+    }
+    
+    
+    x_group <- data %>% 
+        group_by(.data$X) %>% 
+        summarise(group_n = n()) %>%
+        mutate(group_freq = .data$group_n/sum(.data$group_n),
+               x = cumsum(.data$group_freq),
+               xmin = c(0,.data$x[-length(.data$x)]))
+    data <- data %>%
+        group_by(.data$X, .data$Y) %>%
+        summarise(fill_n = n(), .groups = "rowwise") %>%
+        left_join(x_group, by = "X") %>%
+        mutate(fill_freq = .data$fill_n/.data$group_n) %>%
+        group_by(.data$X) %>%
+        mutate(y = cumsum(.data$fill_freq),
+               ymin = c(0,.data$y[-length(.data$y)])) %>%
+        ungroup()
+    data
+}
+
+.plot_tile_data <- function(object,
+                            type = c("row", "column"),
+                            x,
+                            y,
+                            factor_x_by,
+                            factor_y_by,
+                            ...){
+    type <- match.arg(type)
+    tile_out <- .get_tile_data(object, type, x, y)
+    tile_data <- tile_out$data
+    xlab <- tile_out$x_lab
+    ylab <- tile_out$y_lab
+    tile_data <- .summarise_tile_data(tile_data,
+                                      type,
+                                      factor_x_by,
+                                      factor_y_by)
+    tile_data$colour_by <- tile_data$Y
+    .tile_plotter(tile_data,
+                  xlab = xlab,
+                  ylab = ylab,
+                  ...)
+}
+
+.get_xcoord_mid <- function(data){
+    data %>% 
+        ungroup() %>%
+        select(.data$X, .data$x, .data$xmin) %>% 
+        unique() %>%
+        mutate(xmid = .data$xmin + (.data$x - .data$xmin)/2 )
+}
+
+.tile_plotter <- function(data,
+                          add_legend = TRUE,
+                          xlab,
+                          ylab,
+                          rect_alpha = 1,
+                          rect_colour = "black"){
+    coord <- .get_xcoord_mid(data)
+    # get plotting arguments for rect
+    rect_args <- .get_rect_args(colour_by = ylab, 
+                                alpha = rect_alpha,
+                                colour = rect_colour)
+    rect_args$args$mapping$xmin <- sym("xmin")
+    rect_args$args$mapping$xmax <- sym("x")
+    rect_args$args$mapping$ymin <- sym("ymin")
+    rect_args$args$mapping$ymax <- sym("y")
+    # start plotting
+    plot_out <- ggplot(data) 
+    plot_out <- plot_out +
+        do.call(geom_rect,rect_args$args)
```

Comment:
Should `NA` be given color? EDIT: I see the color now its is a shade of very light grey?
```
library(microbiomeDataSets)
library(miaViz)
tse <- SprockettTHData()
plotColTile(tse,"Age_Class","Community_Subtype")
```

Reply:
Add a `na.value` setting to customize the color.